app/assets/javascripts/admin/views/team/create_team_modal_view.js

```diff
@@ -1,74 +1,103 @@
-import _ from "lodash";
-import TeamModel from "admin/models/team/team_model";
-import TeamCollection from "admin/models/team/team_collection";
-import SelectionView from "admin/views/selection_view";
-import ModalView from "admin/views/modal_view";
+import * as React from "react";
+import { Modal, Input, Select, Spin } from "antd";
+import Request from "libs/request";
+import type { APITeamType } from "admin/api_flow_types";
 
-class CreateTeamModalView extends ModalView {
-  static initClass() {
-    this.prototype.headerTemplate = "<h3>Add a New Team</h3>";
-    this.prototype.bodyTemplate = _.template(`\
-<form class="form-horizontal">
-  <div class="form-group">
-    <label class="col-sm-2 control-label" for="inputName">Name</label>
-    <div class="col-sm-10">
-      <input type="text" class="form-control" id="inputName" placeholder="Name" required autofocus>
-    </div>
-  </div>
-  <div class="form-group">
-    <label class="col-sm-2 control-label" for="inputName">Parent Team</label>
-    <div class="col-sm-10 parent-teams">
-      <input type="text" class="form-control" id="" placeholder="Name" required>
-    </div>
-  </div>
-</form>\
-`);
-    this.prototype.footerTemplate = `\
-<button type="submit" class="btn btn-primary">Add</button>
-<a href="#" class="btn btn-default" data-dismiss="modal">Close</a>\
-`;
+const { Option } = Select;
 
-    this.prototype.ui = { inputName: "#inputName" };
+type Props = {
+  onOk: Function,
+  onCancel: Function,
+  visible: boolean,
+};
 
-    this.prototype.events = {
-      "submit form": "addNewTeam",
-      "click .btn-primary": "addNewTeam",
-    };
+type State = {
+  newTeamName: string,
+  parentTeam: ?string,
+  visible: boolean,
+  teams: Array<APITeamType>,
+  isLoading: boolean,
+};
 
-    this.prototype.regions = { parentTeams: ".parent-teams" };
+class CreateTeamModalView extends React.PureComponent<Props, State> {
+  state = {
+    newTeamName: "",
+    parentTeam: undefined,
+    teams: [],
+    isLoading: true,
+  };
+
+  componentDidMount() {
+    this.fetchData();
   }
 
-  initialize(options) {
-    this.teamCollection = options.teamCollection;
+  async fetchData() {
+    const url = "/api/teams?isRoot=true";
+    const teams = await Request.receiveJSON(url);
 
-    this.teamSelectionView = new SelectionView({
-      collection: new TeamCollection(),
-      childViewOptions: {
-        modelValue() {
-          return `${this.model.get("name")}`;
-        },
-      },
-      data: "isRoot=true",
+    this.setState({
+      isLoading: false,
+      teams,
     });
   }
 
-  addNewTeam(evt) {
-    evt.preventDefault();
+  onOk = () => {
+    if (this.state.newTeamName !== "" && this.state.parentTeam !== undefined) {
+      const newTeam = {
+        name: this.state.newTeamName,
+        parent: this.state.parentTeam,
+        roles: [{ name: "admin" }, { name: "user" }],
```

Comment:
The roles cannot be changed (as before) and are also not displayed anymore, right? This is intended, since more powerful role management is not needed, right?

Reply:
Correct, the roles were never displayed and can not be changed just as before. So far, there was no need to go beyond `user` and `admin`roles.